pkg/admission/validatingwebhook/plugin.go

```diff
@@ -0,0 +1,182 @@
+	p := &Plugin{Plugin: validating.Plugin{Webhook: &generic.Webhook{}}}
+	p.Handler = admission.NewHandler(admission.Connect, admission.Create, admission.Delete, admission.Update)
+
+	dispatcherFactory := validating.NewValidatingDispatcher(&p.Plugin)
+
+	// Making our own dispatcher so that we can control the webhook accessors.
+	kubeconfigFile, err := config.LoadConfig(configfile)
+	if err != nil {
+		return nil, err
+	}
+	cm, err := webhookutil.NewClientManager(
+		[]schema.GroupVersion{
+			admissionv1beta1.SchemeGroupVersion,
+			admissionv1.SchemeGroupVersion,
+		},
+		admissionv1beta1.AddToScheme,
+		admissionv1.AddToScheme,
+	)
+	if err != nil {
+		return nil, err
+	}
+	authInfoResolver, err := webhookutil.NewDefaultAuthenticationInfoResolver(kubeconfigFile)
+	if err != nil {
+		return nil, err
+	}
+	// Set defaults which may be overridden later.
+	cm.SetAuthenticationInfoResolver(authInfoResolver)
+	cm.SetServiceResolver(webhookutil.NewDefaultServiceResolver())
+
+	p.dipatcher = dispatcherFactory(&cm)
+	// Need to do this, to make sure that the underlying objects for the call to ShouldCallHook have the right values
+	p.Plugin.Webhook, err = generic.NewWebhook(p.Handler, configfile, configuration.NewValidatingWebhookConfigurationManager, dispatcherFactory)
+	if err != nil {
+		return nil, err
+	}
+
+	//Override the ready func
+
+	p.SetReadyFunc(func() bool {
+		if p.hookSource.HasSynced() && p.apiBindHasSynced() && p.Plugin.WaitForReady() {
+			return true
+		}
+		return false
+	})
+	klog.V(0).Infof("HEREREER~!")
+	fmt.Printf("HERE!!!!")
+	return p, nil
+}
+
+func Register(plugins *admission.Plugins) {
+	plugins.Register(PluginName, func(configFile io.Reader) (admission.Interface, error) {
+		return NewValidatingAdmissionWebhook(configFile)
+	})
+}
+
+func (a *Plugin) Validate(ctx context.Context, attr admission.Attributes, o admission.ObjectInterfaces) error {
+	return a.Dispatch(ctx, attr, o)
+}
+
+func (p *Plugin) Dispatch(ctx context.Context, attr admission.Attributes, o admission.ObjectInterfaces) error {
+	if rules.IsWebhookConfigurationResource(attr) {
+		return nil
+	}
+	lcluster, err := genericapirequest.ClusterNameFrom(ctx)
+	if err != nil {
+		return err
+	}
+	if !p.WaitForReady() {
+		return admission.NewForbidden(attr, fmt.Errorf("not yet ready to handle request"))
+	}
+
+	hooks := p.hookSource.Webhooks()
+	whAccessor := p.restrictToLogicalCluster(hooks, lcluster)
+
+	// Determine the type of request, is it api binding or not.
+	if workspace, isAPIBinding := p.getAPIBindingWorkspace(attr, lcluster); isAPIBinding {
+		wh := p.restrictToLogicalCluster(hooks, workspace)
+		whAccessor = append(whAccessor, wh...)
+	}
+
+	return p.dipatcher.Dispatch(ctx, attr, o, whAccessor)
+}
+
+func (p *Plugin) getAPIBindingWorkspace(attr admission.Attributes, lc logicalcluster.LogicalCluster) (logicalcluster.LogicalCluster, bool) {
+	l, err := p.apiBinding.List(labels.Everything())
```

Comment:
We need something more clever here, something to lookup the information without iterating over 1000 apibindings. An informer index to filter by workspace would help a lot.

Reply:
Can we TODO pending #759 ?